lib/topological_inventory/ansible_tower/operations/source.rb

```diff
@@ -11,8 +11,13 @@ class Source < TopologicalInventory::Providers::Common::Operations::Source
         private
 
         def connection_check
-          connection = ::TopologicalInventory::AnsibleTower::Cloud::Connection.new
-          connection = connection.connect(full_hostname(endpoint), authentication.username, authentication.password)
+          connection = ::TopologicalInventory::AnsibleTower::ConnectionManager.new(source_id).connect(
```

Comment:
I'm not sure if it'll work in all cases. Either make if/else there, or add `.present?` to checks there: https://github.com/RedHatInsights/topological_inventory-ansible_tower/blob/f42cda8cbd05b9360622ba683c5814177a4d5dcb/lib/topological_inventory/ansible_tower/connection_manager.rb#L47